sdk/core/azure-core/CHANGELOG.md

```diff
@@ -13,7 +13,7 @@
 - Added `GetPort()` to `Url`.
 - Added `TransportPolicyOptions`.
 - Added `TelemetryPolicyOptions`.
-- Added `DateFormat::ToIso8601String(TimeFractionFormat)`.
+- Added `DateFormat::ToRfc3339String(TimeFractionFormat)`.
```

Comment:
I would expect we have ToString with an overload that takes a format.  Is there a reason to have a specific method named for this?

Reply:
It's because we only support this for RFC 3339. Other RFC doesn't even have fractional seconds.

Reply:
Why don't we have just a single method, and throw for format enum values that don't mix-match well? Having different named methods like this doesn't scale if and when we add more format support and options for formatting.

```C#
// If the user provides incompatible arguments - validate them and throw:
GetString(DateFormat, TimeFractionFormat);
```

Ideally, we would just have one options paramater:
```C#
GetString(DateFormatOptions = Some Default Value);
```

Reply:
Why throw, if the exception can be prevented in compile-time?
But let's discuss the formatting overall, later.

Reply:
>  this doesn't scale if and when we add more format support

What if adding another date format now also support/requires other specific TimeFractionFormat or other type of options?
Will you add one extra parameter to `GetString` and now validate the combination of 3 parameters?
The way that escalates is nasty.

Agree with Anton, if we can prevent things that shouldn't be called from compilation time, let's do it like that. Otherwise, what's the benefit giving valid enum input that would do nothing but fail?

Reply:
> Why throw, if the exception can be prevented in compile-time?

Trade-off to avoid API explosion.

> What if adding another date format now also support/requires other specific TimeFractionFormat or other type of options?

This is why the single struct argument that captures all future customizations is most extensible. We have this pattern elsewhere in the SDK, so customers should be familiar with it.

What would you do in that case? Add more differently named methods taking various other enums and options? So, the DateTime will have a growing list of GetXStringY(some parameter) APIs?

Reply:
I'm not opposed to having full formatString support; But I realize it is likely never going to be used.

Reply:
Jeffrey opposed to having default value, at least in this specific case.

I don't think we are going to reach an agreement on this today, more discussions are needed. I unresolved all the comments that might provide food for discussions / info for the future - on this topics and others.

I am going to complete this PR. After all, GetRfc3339String() is not a breaking change, should we want to make `GetString(DateFormat, Fractionaloptions)` public in the future. Maybe it won't look as elegant, but technically we even still can be making breaking changes in this stage.